bench/datasets/test_dataset.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import json
+import os
+
+from tools.nvt_etl import nvt_etl
+
+from nvtabular import Workflow
+
+
+def main(args):
+    # Get cats/conts/labels
+    with open(args.config_file) as f:
+        data = json.load(f)
+    cats = data["cats"]
+    conts = data["conts"]
+    labels = data["labels"]
+
+    # Perform ETL
+    if not args.workflow_path:
+        workflow = nvt_etl(
+            args.data_path,
+            args.out_path,
+            args.devices,
+            args.protocol,
+            args.device_limit_frac,
+            args.device_pool_frac,
+            args.part_mem_frac,
+            cats,
+            conts,
+            labels,
+            args.out_files_per_proc,
+        )
+    else:
+        workflow = Workflow.load(os.path.join(args.workflow_path, "workflow"))
+
+    # Perform training
+    # HugeCTR
+    if args.target_framework == "hugectr":

```

Comment:
So I looked into this several times. So if we use a dictionary of functions and we come up with this abstraction, we don't know which function we are using. If you see, I am only importing the function files once I know which target framework I am using. This is because that function has imports in its file that require different frameworks, so it will produce an error because it depends on the container we have one package or another. @jperez999 does it makes sense?

Reply:
I see your point. We can leave it as is.